README.md

```diff
@@ -7,14 +7,78 @@ These features can be used to implement expressive class hierarchies as well as
 
 MLscript supports union, intersection, and complement (or negation) connectives, making sure they form a Boolean algebra, and add enough structure to derive a sound and complete type inference algorithm.
 
+## Getting Started Guide
 
-## Running the tests
+### Project Structure
 
-Running the tests only requires the Scala Build Tool installed.
+#### Sub-Projects
+
+- The ts2mls sub-project allows you to use TypeScript libraries in MLScript. It can generate libraries' declaration information in MLScript by parsing TypeScript AST, which can be used in MLScript type checking.
+
+#### Directories
+
+- The `shared/src/main/scala/mlscript/` directory contains the sources of the MLscript compiler.
+
+- The `shared/src/test/scala/mlscript` directory contains the sources of the testing infrastructure.
+
+- The `shared/src/test/diff` directory contains the actual tests.
+
+- The `ts2mls/js/src/main/scala/ts2mls` directory contains the sources of the ts2mls module.
+
+- The `ts2mls/js/src/test/scala/ts2mls` directory contains the sources of the ts2mls declaration generation test code.
+
+- The `ts2mls/jvm/src/test/scala/ts2mls` directory contains the sources of the ts2mls diff test code.
+
+- The `ts2mls/js/src/test/typescript` directory contains the TypeScript test code.
+
+- The `ts2mls/js/src/test/diff` directory contains the declarations generated by ts2mls.
+
+### Prerequisites
+
+To compile the project and run the tests, all you need is a recent Java Virtual Machine (JVM), the Scala Build Tool (SBT), NodeJS, and TypeScript.
+
+The simplest way to set up the JVM and SBT is through the `coursier` tool.
+We explicitly support NodeJS versions `v16.14.x`, `v16.15.x`, `v16.16.x`, and `v17.x.x`.
```

Comment:
We also tested `v18` if I remembered correctly?

Reply:
I copied this information from [getting-started.md](https://github.com/hkust-taco/mlstruct/blob/mlstruct/getting-started.md). My node's version is also `17.x.x` so I'm not sure. :joy:

Reply:
Just look at NodeTest.scala. I think it now supports it.

Reply:
Yes indeed.